src/settings/SettingsArea.tsx

```diff
@@ -61,68 +64,195 @@ const SettingsArea = () => {
 
   return (
-    <VStack alignItems="flex-start" padding={3} spacing={5}>
-      <FormControl display="flex" alignItems="center">
-        <FormLabel
-          htmlFor="font-size"
-          mb="0"
-          fontWeight="normal"
-          flex="1 1 auto"
-        >
-          <FormattedMessage id="font-size" />
-        </FormLabel>
-        <NumberInput
-          id="font-size"
-          size="sm"
-          value={settings.fontSize}
-          min={minimumFontSize}
-          max={maximumFontSize}
-          onChange={handleChangeFontSize}
-          width="12ch"
-        >
-          <NumberInputField />
-          <NumberInputStepper>
-            <NumberIncrementStepper />
-            <NumberDecrementStepper />
-          </NumberInputStepper>
-        </NumberInput>
-      </FormControl>
-      <FormControl display="flex" alignItems="center">
-        <FormLabel
-          htmlFor="language"
-          mb="0"
-          fontWeight="normal"
-          flex="1 1 auto"
-        >
-          <FormattedMessage id="highlight-structure" />
-        </FormLabel>
-        <Select
-          id="language"
-          variant="outline"
-          onChange={handleChangeCodeStructureHighlight}
-          width="20ch"
-          value={settings.codeStructureHighlight}
-        >
-          {codeStructureHighlightOptions.map(({ value, label }) => (
-            <option key={value} value={value}>
-              {label}
-            </option>
-          ))}
-        </Select>
-      </FormControl>
+    <VStack alignItems="flex-start" spacing={3}>
+      <VStack alignItems="flex-start" padding={3} spacing={5}>
+        <Text as="h2" fontWeight="semibold">
+          Common
+        </Text>
+        <FormControl display="flex" alignItems="center">
+          <FormLabel
+            htmlFor="font-size"
+            mb="0"
+            fontWeight="normal"
+            flex="1 1 auto"
+          >
+            <FormattedMessage id="font-size" />
+          </FormLabel>
+          <NumberInput
+            id="font-size"
+            size="sm"
+            value={settings.fontSize}
+            min={minimumFontSize}
+            max={maximumFontSize}
+            onChange={handleChangeFontSize}
+            width="12ch"
+          >
+            <NumberInputField />
+            <NumberInputStepper>
+              <NumberIncrementStepper />
+              <NumberDecrementStepper />
+            </NumberInputStepper>
+          </NumberInput>
+        </FormControl>
+      </VStack>
+      <VStack alignItems="flex-start" padding={3} spacing={5}>
+        <Text as="h2" fontWeight="semibold">
+          Code structure highlighting (experimental)
+        </Text>
+        <SelectFormControl
+          id="codeStructureShape"
+          label="Shape"
+          options={[
+            {
+              value: "l-shape",
+              label: "L-shapes",
+            },
+            {
+              value: "box",
+              label: "Boxes",
+            },
```

Comment:
The text here and throughout could be extracted for i18n but not sure it's worth it for these experimental settings. The "Common" heading maybe should be, once we're happy with the text?